### PR TITLE
Feature: logs2notifications

### PR DESCRIPTION
**Pull request by shreddedbacon**
# Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [ ] PR title is ready for changelog and subsystem label(s) applied

This change adds logs2notifications service to Lagoon. It replaces the existing logs2slack, logs2rocketchat, logs2microsoftteams, logs2webhooks, and logs2email services.

Currently the logs2x services all consume from individual queues, each service will then check the API for its specific notification types for the project/environment the logs is associated to, and then process the message according. This is currently 5 API requests for each message received, adding unnecessary load on the API in a busy Lagoon.

The new notifications system does this a little bit differently and processes from one queue only, then checks the API for each message for all the notification types available to that project, this is just one API request. Then it sends the message to the appropriate message handler service within logs2notifications.

The `logs2s3` system is also consolidated into the system as it functions in a similar way to other notifications systems, while not being notifications like slack, or the other types.

### CODE REVIEW

**Comment by tobybellwood** [comment]
We'll need to make sure that these workspaces are removed from yarn.lock

**Comment by rocketeerbkw** [comment]
Instructions for cleaning up yarn https://github.com/uselagoon/lagoon/pull/1798#issuecomment-616115806

**Comment by shreddedbacon** [comment]
This has been running in the test infrastructure for a couple of days and everything is working as expected with notifications :partying_face:

File: services/logs2notifications/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# build the binary
+ARG GO_VERSION
+ARG UPSTREAM_REPO
+ARG UPSTREAM_TAG
+FROM golang:${GO_VERSION:-1.16.4} AS builder
```
Comment (tobybellwood) [inline services/logs2notifications/Dockerfile]: we just standardized all the other go services to use `FROM golang:1.18-alpine3.16 AS builder` - can this be easy updated to go 1.18?

Comment (shreddedbacon) [reply services/logs2notifications/Dockerfile]: Probably, will fix it up

**Review by tobybellwood — APPROVED** [review]
this works - lets begin the process to get it merged!

**Pull request merged**